src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.spec.tsx

```diff
@@ -22,43 +25,84 @@ describe('<Enrolled/>', () => {
     await screen.findByText(
       'You are enrolled for the session from ' +
-        new Intl.DateTimeFormat('en', DATETIME_FORMAT).format(
+        new Intl.DateTimeFormat('en', DEFAULT_DATE_FORMAT).format(
           new Date(enrollment.course_run.start),
         ) +
         ' to ' +
-        new Intl.DateTimeFormat('en', DATETIME_FORMAT).format(new Date(enrollment.course_run.end)),
+        new Intl.DateTimeFormat('en', DEFAULT_DATE_FORMAT).format(
+          new Date(enrollment.course_run.end),
+        ),
     );
     if (expectButton) {
-      const link = screen.getByRole('link', { name: 'Access course' });
+      const link = screen.getByRole('link', { name: 'Access to course' });
       expect(link).toBeEnabled();
       expect(link).toHaveAttribute('href', enrollment.course_run.resource_link);
     } else {
-      expect(screen.queryByRole('link', { name: 'Access course' })).toBeNull();
+      expect(screen.queryByRole('link', { name: 'Access to course' })).toBeNull();
     }
   };
 
-  it('handles enrollments with priority=ONGOING_OPEN', () => {
-    runTest(Priority.ONGOING_OPEN, true);
+  it('handles enrollments with priority=ONGOING_OPEN', async () => {
+    await runTest(Priority.ONGOING_OPEN, true);
   });
-  it('handles enrollments with priority=FUTURE_OPEN', () => {
-    runTest(Priority.FUTURE_OPEN, false);
+  it('handles enrollments with priority=FUTURE_OPEN', async () => {
+    await runTest(Priority.FUTURE_OPEN, false);
   });
-  it('handles enrollments with priority=ARCHIVED_OPEN', () => {
-    runTest(Priority.ARCHIVED_OPEN, true);
+  it('handles enrollments with priority=ARCHIVED_OPEN', async () => {
+    await runTest(Priority.ARCHIVED_OPEN, true);
   });
-  it('handles enrollments with priority=FUTURE_NOT_YET_OPEN', () => {
-    runTest(Priority.FUTURE_NOT_YET_OPEN, false);
+  it('handles enrollments with priority=FUTURE_NOT_YET_OPEN', async () => {
+    await runTest(Priority.FUTURE_NOT_YET_OPEN, false);
   });
-  it('handles enrollments with priority=FUTURE_CLOSED', () => {
-    runTest(Priority.FUTURE_CLOSED, false);
+  it('handles enrollments with priority=FUTURE_CLOSED', async () => {
+    await runTest(Priority.FUTURE_CLOSED, false);
   });
-  it('handles enrollments with priority=ONGOING_CLOSED', () => {
-    runTest(Priority.ONGOING_CLOSED, true);
+  it('handles enrollments with priority=ONGOING_CLOSED', async () => {
+    await runTest(Priority.ONGOING_CLOSED, true);
   });
-  it('handles enrollments with priority=ARCHIVED_CLOSED', () => {
-    runTest(Priority.ARCHIVED_CLOSED, true);
+  it('handles enrollments with priority=ARCHIVED_CLOSED', async () => {
+    await runTest(Priority.ARCHIVED_CLOSED, true);
   });
-  it('handles enrollments with priority=TO_BE_SCHEDULED', () => {
-    runTest(Priority.TO_BE_SCHEDULED, false);
+  it('handles enrollments with priority=TO_BE_SCHEDULED', async () => {
+    await runTest(Priority.TO_BE_SCHEDULED, false);
```

Comment:
it could also be a each like 

```ts
it.each([
    {
      buttonTestlabel: 'and access course button',
      priority: Priority.ONGOING_OPEN,
      expectButton: true,
    },
    {
      buttonTestlabel: 'and no access course button',
      priority: Priority.FUTURE_OPEN,
      expectButton: false,
    },
    {
      buttonTestlabel: 'and access course button',
      priority: Priority.ARCHIVED_OPEN,
      expectButton: true,
    },
    {
      buttonTestlabel: 'and no access course button',
      priority: Priority.FUTURE_NOT_YET_OPEN,
      expectButton: false,
    },
    {
      buttonTestlabel: 'and no access course button',
      priority: Priority.FUTURE_CLOSED,
      expectButton: false,
    },
    {
      buttonTestlabel: 'and access course button',
      priority: Priority.ONGOING_CLOSED,
      expectButton: true,
    },
    {
      buttonTestlabel: 'and access course button',
      priority: Priority.ARCHIVED_CLOSED,
      expectButton: true,
    },
    {
      buttonTestlabel: 'and no access course button',
      priority: Priority.TO_BE_SCHEDULED,
      expectButton: false,
    },
  ])(
    'handles enrollments with priority=$priority $buttonTestlabel',
    async ({ priority, expectButton }) => {
      const enrollment: Enrollment = EnrollmentFactory().one();
      enrollment.course_run.state.priority = priority;
      render(<Enrolled enrollment={enrollment} />, { wrapper });
      await screen.findByText(
        'You are enrolled for the session from ' +
          new Intl.DateTimeFormat('en', DATETIME_FORMAT).format(
            new Date(enrollment.course_run.start),
          ) +
          ' to ' +
          new Intl.DateTimeFormat('en', DATETIME_FORMAT).format(
            new Date(enrollment.course_run.end),
          ),
      );
      if (expectButton) {
        const link = screen.getByRole('link', { name: 'Access course' });
        expect(link).toBeEnabled();
        expect(link).toHaveAttribute('href', enrollment.course_run.resource_link);
      } else {
        expect(screen.queryByRole('link', { name: 'Access course' })).toBeNull();
      }
    },
  );
```

Reply:
Good idea :)